examples/vector_databases/redis/getting-started-with-redis-and-openai.ipynb

```diff
@@ -0,0 +1,809 @@
+    "3. Create the index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "id": "f894b911",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Constants\n",
+    "VECTOR_DIM = len(data['title_vector'][0]) # length of the vectors\n",
+    "VECTOR_NUMBER = len(data)                 # initial number of vectors\n",
+    "INDEX_NAME = \"embeddings-index\"           # name of the search index\n",
+    "PREFIX = \"doc\"                            # prefix for the document keys\n",
+    "DISTANCE_METRIC = \"COSINE\"                # distance metric for the vectors (ex. COSINE, IP, L2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 85,
+   "id": "15db8380",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define RediSearch fields for each of the columns in the dataset\n",
+    "title = TextField(name=\"title\")\n",
+    "url = TextField(name=\"url\")\n",
+    "text = TextField(name=\"text\")\n",
+    "title_embedding = VectorField(\"title_vector\",\n",
+    "    \"FLAT\", {\n",
+    "        \"TYPE\": \"FLOAT32\",\n",
+    "        \"DIM\": VECTOR_DIM,\n",
+    "        \"DISTANCE_METRIC\": DISTANCE_METRIC,\n",
+    "        \"INITIAL_CAP\": VECTOR_NUMBER,\n",
+    "    }\n",
+    ")\n",
+    "text_embedding = VectorField(\"content_vector\",\n",
+    "    \"FLAT\", {\n",
+    "        \"TYPE\": \"FLOAT32\",\n",
+    "        \"DIM\": VECTOR_DIM,\n",
+    "        \"DISTANCE_METRIC\": DISTANCE_METRIC,\n",
+    "        \"INITIAL_CAP\": VECTOR_NUMBER,\n",
+    "    }\n",
+    ")\n",
+    "fields = [title, url, text, title_embedding, text_embedding]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 86,
+   "id": "3658693c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check if index exists\n",
+    "try:\n",
+    "    redis_client.ft(INDEX_NAME).info()\n",
+    "    print(\"Index already exists\")\n",
+    "except:\n",
+    "    # Create RediSearch Index\n",
+    "    redis_client.ft(INDEX_NAME).create_index(\n",
+    "        fields = fields,\n",
+    "        definition = IndexDefinition(prefix=[PREFIX], index_type=IndexType.HASH)\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "775c15b4",
+   "metadata": {},
+   "source": [
+    "## Load Documents into the Index\n",
+    "\n",
+    "Now that we have a search index, we can load documents into it. We will use the same documents we used in the previous examples. In Redis, either the Hash or JSON (if using RedisJSON in addition to RediSearch) data types can be used to store documents. We will use the HASH data type in this example. The below cells will show how to load documents into the index."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 87,
+   "id": "0d791186",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def index_documents(client: redis.Redis, prefix: str, documents: pd.DataFrame):\n",
```

Comment:
Could be good to at least show simple `pipeline` usage here since the synchronous function can take a while? Batches of 100-200 seem to be a sweet spot

Reply:
I'm planning a future "when to use Async/pipelines" notebook which this will go in. Dataset is small enough right now to where it only takes a minute to load in everything